TitanRobotics2024/src/main/java/frc/robot/Auto/Actions/DriveForDistanceAction.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.Auto.Actions;
+
+import frc.robot.Subsystem.DriveBase;
+import frc.robot.Subsystem.PositionEstimation;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+
+public class DriveForDistanceAction implements Actions 
+{
+    Timer timer;
+    private double currentDistance;
+    private double desiredDistance;
+    private double targetDistance;
+    private double forward;
+    private double toleranceDistance = 0.0025;
+    private double endAfterSeconds;
+
+    private DriveBase driveBase;
+    private PositionEstimation position;
+
+    private PIDController PID;
+    private double kp = 0.01;
```

Comment:
I assume you want these three variables to be unchangeable, might want to make them all final
```suggestion
    private final double kp = 0.01;
```

Reply:
done